src/cdk/overlay/overlay-config.ts

```diff
@@ -59,6 +59,16 @@ export class OverlayConfig {
    */
   disposeOnNavigation?: boolean = false;
 
+  /**
+   * Whether the overlay should be detached on outside click
+   */
+  detachOnOutsideClick?: boolean = false;
+
+  /**
+   * Array of HTML elements clicking on which should not be considered as outside click
+   */
+  excludeFromOutsideClick?: HTMLElement[] = [];
```

Comment:
I'm not quite sure what the use case is for this.

Reply:
This array will contain elements which will be excluded from outside clicks. Let say you have two buttons. Clicking on each one will open a mat-menu. Right now if you click on button one mat-menu will be shown. Then if you click on button two mat-menu will be closed, but second mat-menu will not be shown. As described in [this](https://github.com/angular/components/issues/9320) issue.
If we set `detachOnOutsideClick` for mat-menu to `true` this will work (as well as set `hasBackdrop` to false). However in this case when you click on button one mat-menu will be shown. If at this point you click again at button one this will happen:
1. Document's body will recognize this as outside click and will force mat-menu to close.
2. Click event will reach the button one. At this point the mat-menu is already closed, so the button will try to open it.
To prevent this from happen we should say to overlay consider clicking on the button as not outside click. And we can achieve this by adding the  button to `excludeFromOutsideClick` array.

Reply:
Now that `detachOnOutsideClick` has been removed in favor of having a click stream, is this option still necessary? Consumers of the API can now make the determination of what to do on the click based on what was clicked, right?

Reply:
I was thinking for the situations where `overlayRef` has no clue who opened it, and one could not determine in the handler of `outsideClickEvents`, clicking on which element on the page should or should not detach the overlay.